### ISSUE TITLE
Update ECMPS to import cpms [MATS] qualifications (mp)

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#5396]
## Context 

As apart of the scope of this ticket ensure users can import the new schema for cpms qualifications (mp) data into the system 

[BETA Web Client MATS 2.0 MP Reporting Instructions June 2023.docx](https://usepa.sharepoint.com/:w:/r/sites/CAMDITLegacyReeng/_layouts/15/Doc.aspx?sourcedoc=%7B52DA5270-E093-4DCF-AC7C-124A253A721C%7D&file=BETA%20Web%20Client%20MATS%202.0%20MP%20Reporting%20Instructions%20June%202023.docx&wdLOR=c92FC035F-4657-4B3F-95FA-8BA07BAEAD9D&action=default&mobileredirect=true)

[BETA Web Client MATS 2.0 EM Reporting Instructions June ](https://usepa.sharepoint.com/:w:/r/sites/CAMDITLegacyReeng/_layouts/15/Doc.aspx?sourcedoc=%7BFE3CF8DF-B3A9-4485-83D6-993FECFD5EC1%7D&file=BETA%20Web%20Client%20MATS%202.0%20EM%20Reporting%20Instructions%20June%202023.docx&wdLOR=cE98A897F-0D9F-4BC0-88CF-CF9C649B997F&action=default&mobileredirect=true)

[BETA Web Client MATS 2.0 EM Reporting Instructions June 2023.docx](https://usepa.sharepoint.com/:w:/r/sites/CAMDITLegacyReeng/_layouts/15/Doc.aspx?sourcedoc=%7BFE3CF8DF-B3A9-4485-83D6-993FECFD5EC1%7D&file=BETA%20Web%20Client%20MATS%202.0%20EM%20Reporting%20Instructions%20June%202023.docx&wdLOR=cF2AC24DB-6DDC-4525-B8B7-19C8FDD210E5&action=default&mobileredirect=true)

### **Schema**
```
"CpmsQualification": {
      "title": "CpmsQualification",
      "type": "object",
      "additionalProperties": true,
      "properties": {
        "qualificationDataYear": {
          "type": "number",
          "minimum": 1900,
          "maximum": 2099
        },
        "stackTestNumber": {
          "type": "string"
        },        
        "operatingLimit": {
          "type": "number",
          "multipleOf": 0.1,
          "minimum": -99999.9,
          "maximum": 99999.9
        }
      },
      "required": [
        "qualificationDataYear",
        "stackTestNumber",
        "operatingLimit"
      ]
    },
```
### **Import JSON**
```
{
        "qualificationDataYear": 2022,
        "stackTestNumber": "Test1234",
        "operatingLimit": 99999.9
},
```
### **Export JSON**
```
{
        "id": "string",
        "qualificationId": "string",
        "qualificationDataYear": 2022,
        "stackTestNumber": "Test1234",
        "operatingLimit": 99999.9
        "userId": "bvick",
        "addDate": "2023-01-20T22:40:24.000Z",
        "updateDate": "2023-01-20T22:40:24.000Z",
    },
```
### **DB Table**
```
CREATE TABLE IF NOT EXISTS camdecmps.monitor_qualification_cpms(
  mon_qual_cpms_id character varying(45) NOT NULL,
  mon_qual_id character varying(45) NOT NULL,
	qual_data_year numeric(4,0) NOT NULL,
	stack_test_number character varying(18) NOT NULL,
	operating_limit numeric(6,1) NOT NULL,
  userid character varying(25) NOT NULL,
  add_date timestamp without time zone,
  update_date timestamp without time zone,
  CONSTRAINT pk_monitor_qualification_cpms PRIMARY KEY (mon_qual_cpms_id),
  CONSTRAINT fk_monitor_qualification_cpms_monitor_qualification FOREIGN KEY (mon_qual_id)
      REFERENCES camdecmps.monitor_qualification (mon_qual_id) MATCH SIMPLE
      ON DELETE CASCADE
);
```

### Communications 
Also as part of this task, can you create one ticket to create the structures, and another ticket to update the beta to accept the import of these versions?  

From: Worley, Christopher 
Sent: Friday, June 23, 2023 11:10 AM
To: jasonwhitehead@cvpcorp.com; Kyle Herceg <kyleherceg@cvpcorp.com>; Whitten, Dwayne <whitten.dwayne@epa.gov>
Cc: Alex Collado <alexcollado@cvpcorp.com>; janell coleman <janellcoleman@cvpcorp.com>; Heese, Michael <Heese.Michael@epa.gov>; Ibarra, Michelle <Ibarra.Michelle@epa.gov>
Subject: FW: Updated Instructions

Jason, 

The updated instructions for MP and EM are posted and complete.  Let me know what else you need to create the updated JSON schema documentation.

Thanks.

Chris Worley
202-816-2670

From: Sarah Downes <Sarah.Downes@erg.com> 
Sent: Thursday, June 22, 2023 5:03 PM
To: Worley, Christopher <worley.christopher@epa.gov>; Frushour, Charles <Frushour.Charles@epa.gov>
Cc: Scott Aber <scott.aber@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>; Dwayne Whitten <dwayne.whitten@erg.com>; Lane Giannini <lane.giannini@erg.com>; Schreifels, Jeremy <Schreifels.Jeremy@epa.gov>
Subject: RE: Updated Instructions

Chris, 

The update for Daily Backstop has been added to the Emissions instructions on CAMD’s SharePoint.

BETA Web Client MATS 2.0 EM Reporting Instructions June 2023.docx

Sarah Downes
508.340.0198

From: Worley, Christopher <worley.christopher@epa.gov> 
Sent: Thursday, June 1, 2023 3:58 PM
To: Sarah Downes <Sarah.Downes@erg.com>; Frushour, Charles <Frushour.Charles@epa.gov>
Cc: Scott Aber <Scott.Aber@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>; Dwayne Whitten <Dwayne.Whitten@erg.com>; Lane Giannini <Lane.Giannini@erg.com>; Schreifels, Jeremy <Schreifels.Jeremy@epa.gov>
Subject: RE: Updated Instructions

CAUTION: Don't open links or attachments unless you recognize the sender and know they are safe.
Thanks Sarah,  I appreciate all the hard work you and others at ERG have done. 

From: Sarah Downes <Sarah.Downes@erg.com> 
Sent: Thursday, June 1, 2023 5:50 PM
To: Worley, Christopher <worley.christopher@epa.gov>; Frushour, Charles <Frushour.Charles@epa.gov>
Cc: Scott Aber <scott.aber@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>; Dwayne Whitten <dwayne.whitten@erg.com>; Lane Giannini <lane.giannini@erg.com>
Subject: Updated Instructions

Chris, 

The Monitoring Plan and Emissions reporting instructions have been posted to CAMD’s SharePoint:

BETA Web Client MATS 2.0 MP Reporting Instructions June 2023.docx

BETA Web Client MATS 2.0 EM Reporting Instructions June 2023.docx

Thanks,
Sarah

Sarah Downes
Eastern Research Group, Inc.
508.340.0198
sarah.downes@erg.com

**Comment by mosesdeeCVP** [comment]
Import works as expected. User was able to add data and see it in the export

**Comment by mosesdeeCVP** [comment]
![CpmsQualification Import and Export.png](https://images.zenhubusercontent.com/5f807025564a417cbcdf6d69/e375b012-4db8-4147-a4c1-6f233ce47f80)
